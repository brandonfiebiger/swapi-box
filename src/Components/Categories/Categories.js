import React from 'react';
import './Categories.scss';

const Categories = ({ categories, selectedCategory, selectCategory }) => {
  const buttons = categories.map( (category, index) => {
    let selectedBtn = '';

    if (selectedCategory === category) {
      selectedBtn = 'btn-selected';
    }
    
    return <button 
      name={category}
      key={index}
      className={'btn ' + selectedBtn}
      onClick={(e) => selectCategory(e.target.name)}>
      {category}
    </button>
  });

  return (
    <nav className='nav'>{buttons}</nav>
  )
}

export default Categories;