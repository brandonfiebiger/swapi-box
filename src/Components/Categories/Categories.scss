.nav {
 
}

.btn {
  width: 10rem;
  height: 10rem;
  border: 10px solid #763638;
  border-radius: 50%;
  color: #f2f2f2;
  text-shadow: 2px 2px #353535;
  font-size: 2rem;
  outline: none;
}

.btn:hover {
  border: none;
  cursor: pointer;
  color: #f2f2f2;
  text-shadow: 2px 2px #ff0000;
}

.btn:active {
  box-shadow: 3px 5px 10px #888888;
}

.btn-selected {
  border: 10px solid #ff0000;
  text-shadow: 2px 2px #ff0000;
}

.btn-selected:hover {
  border: 10px solid #ff0000;
  text-shadow: 2px 2px #ff0000;
}

.people {
  background: url(../../images/people-btn.jpg) no-repeat center center;
  background-size: 12rem;
}

.planets {
  background: url(../../images/planets-btn.png) no-repeat center center;
  background-size: 11rem;
}

.vehicles {
  background: url(../../images/vehicles-btn.jpg) no-repeat center center;
  background-size: 12rem;
}