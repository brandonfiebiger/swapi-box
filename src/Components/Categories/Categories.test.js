import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';

import Categories from './Categories';

describe('Categories', () => {

  let wrapper;
  let categories;
  let mockFn;

  beforeEach(() => {
    categories = ['word', 'up', 'money'];
    mockFn = jest.fn()
    wrapper = shallow(<Categories categories={categories} selectCategory={mockFn}/>);

  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with one category', () => {
    categories = ['people'];
    expect(wrapper).toMatchSnapshot();

  })

  it('should match snapshot with multiple categories', () => {
    categories = ['people', 'planets', 'vehicles'];
    expect(wrapper).toMatchSnapshot();
  })

  it('should invoke a selectCategory method on click', () => {
    wrapper = mount(<Categories categories={categories} selectCategory={mockFn}/>)
    wrapper.find('button').first().simulate('click')
    expect(wrapper.props().selectCategory).toHaveBeenCalled()

  })

  // it('currentCategory should be highlighted', () => {
  //   wrapper = mount(<Categories categories={categories} selectCategory={mockFn}/>)
  //   wrapper.find('button').first().simulate('click');
  //   expect(wrapper.find('button').first().hasClass('btn-selected')).toEqual(true)
  // })
})