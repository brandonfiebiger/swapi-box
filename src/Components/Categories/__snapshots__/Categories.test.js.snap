// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Categories
    categories={
        Array [
            "word",
            "up",
            "money",
          ]
    }
    selectCategory={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn "
          name="word"
          onClick={[Function]}
>
          word
</button>,
        <button
          className="btn "
          name="up"
          onClick={[Function]}
>
          up
</button>,
        <button
          className="btn "
          name="money"
          onClick={[Function]}
>
          money
</button>,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "word",
          "className": "btn ",
          "name": "word",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "word",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "up",
          "className": "btn ",
          "name": "up",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "up",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "money",
          "className": "btn ",
          "name": "money",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "money",
        "type": "button",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn "
            name="word"
            onClick={[Function]}
>
            word
</button>,
          <button
            className="btn "
            name="up"
            onClick={[Function]}
>
            up
</button>,
          <button
            className="btn "
            name="money"
            onClick={[Function]}
>
            money
</button>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "word",
            "className": "btn ",
            "name": "word",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "word",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "up",
            "className": "btn ",
            "name": "up",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "up",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "money",
            "className": "btn ",
            "name": "money",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "money",
          "type": "button",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Categories should match snapshot with multiple categories 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Categories
    categories={
        Array [
            "word",
            "up",
            "money",
          ]
    }
    selectCategory={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn "
          name="word"
          onClick={[Function]}
>
          word
</button>,
        <button
          className="btn "
          name="up"
          onClick={[Function]}
>
          up
</button>,
        <button
          className="btn "
          name="money"
          onClick={[Function]}
>
          money
</button>,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "word",
          "className": "btn ",
          "name": "word",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "word",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "up",
          "className": "btn ",
          "name": "up",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "up",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "money",
          "className": "btn ",
          "name": "money",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "money",
        "type": "button",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn "
            name="word"
            onClick={[Function]}
>
            word
</button>,
          <button
            className="btn "
            name="up"
            onClick={[Function]}
>
            up
</button>,
          <button
            className="btn "
            name="money"
            onClick={[Function]}
>
            money
</button>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "word",
            "className": "btn ",
            "name": "word",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "word",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "up",
            "className": "btn ",
            "name": "up",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "up",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "money",
            "className": "btn ",
            "name": "money",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "money",
          "type": "button",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Categories should match snapshot with one category 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Categories
    categories={
        Array [
            "word",
            "up",
            "money",
          ]
    }
    selectCategory={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn "
          name="word"
          onClick={[Function]}
>
          word
</button>,
        <button
          className="btn "
          name="up"
          onClick={[Function]}
>
          up
</button>,
        <button
          className="btn "
          name="money"
          onClick={[Function]}
>
          money
</button>,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "word",
          "className": "btn ",
          "name": "word",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "word",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "up",
          "className": "btn ",
          "name": "up",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "up",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "money",
          "className": "btn ",
          "name": "money",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "money",
        "type": "button",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn "
            name="word"
            onClick={[Function]}
>
            word
</button>,
          <button
            className="btn "
            name="up"
            onClick={[Function]}
>
            up
</button>,
          <button
            className="btn "
            name="money"
            onClick={[Function]}
>
            money
</button>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "word",
            "className": "btn ",
            "name": "word",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "word",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "up",
            "className": "btn ",
            "name": "up",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "up",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "money",
            "className": "btn ",
            "name": "money",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "money",
          "type": "button",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
